platform/android/plugin/godot_plugin_jni.cpp

```diff
@@ -114,9 +114,12 @@ JNIEXPORT void JNICALL Java_org_godotengine_godot_plugin_GodotPlugin_nativeEmitS
 	String signal_name = jstring_to_string(j_signal_name, env);
 
 	int count = env->GetArrayLength(j_signal_params);
-	Variant variant_params[count];
-	const Variant *args[count];
+	ERR_FAIL_COND_MSG(count > VARIANT_ARG_MAX, "Maximum argument count exceeded!");
 
+	Variant variant_params[VARIANT_ARG_MAX];
+	const Variant *args[VARIANT_ARG_MAX];
+
+	count = MIN(count, VARIANT_ARG_MAX);
```

Comment:
You don't need to take the `MIN` here as your `ERR_FAIL_COND` above ensures that this code won't be reached if `count > VARIANT_ARG_MAX`.